File: 28.implement-str-str.java
/*
 * @lc app=leetcode id=28 lang=java
 *
 * [28] Implement strStr()
 */

// @lc code=start
class Solution {
    public int strStr(String haystack, String needle) {
        if (haystack.length() < needle.length())
            return -1;
        if (needle.length() == 0)
            return 0;
        
        for(int i=0; i < haystack.length() - needle.length() + 1; i++) {
            if (haystack.charAt(i) == needle.charAt(0)) {
                int j=1;
                for(; j< needle.length(); j++) {
                    if(haystack.charAt(i+j) != needle.charAt(j)) {
                        break;
                    }
                }
                if (j == needle.length())
                    return i;
            }
        }
        return -1;
    }
}
// @lc code=end

File: algorithms/28.implement-str-str.go
/*
 * @lc app=leetcode id=28 lang=golang
 *
 * [28] Implement strStr()
 */

// @lc code=start
func strStr(haystack string, needle string) int {
	m := len(haystack)
	n := len(needle)
	if m < n {
		return -1
	}
	if n == 0 {
		return 0
	}
	match := true
	for i, _ := range haystack[:m-n+1] {
		match = true
		for j, nc := range needle {
			if haystack[i+j] != byte(nc) {
				match = false
				break
			}
		}
		if match {
			return i
		}
	}
	return -1
}

// @lc code=end
